test/node/bigint_tests.ts

```diff
@@ -0,0 +1,123 @@
+      '7856341278563412' // little endian int64
+    ]);
+    expect(serializedDoc).to.deep.equal(expectedResult);
+  });
+
+  it('Correctly serializes a BigInt that can be safely represented as a Number', function () {
+    const testDoc = { a: BigInt(0x23) };
+    const serializedDoc = BSON.serialize(testDoc);
+    const expectedResult = byteUtils.bufferFromHexArray([
+      '12', // int64 type
+      '6100', // 'a' key with key null terminator
+      '2300000000000000' // little endian int64
+    ]);
+    expect(serializedDoc).to.deep.equal(expectedResult);
+  });
+
+  it('Correctly serializes a BigInt in the valid range [-2^63, 2^63 - 1]', function () {
+    const testDoc = { b: BigInt(0xfffffffffffffff1n) };
+    const serializedDoc = BSON.serialize(testDoc);
+    const expectedResult = byteUtils.bufferFromHexArray([
+      '12', // int64
+      '6200', // 'b' '\0'
+      'f1ffffffffffffff'
+    ]);
+    expect(serializedDoc).to.deep.equal(expectedResult);
+  });
+
+  it('Correctly wraps to negative on a BigInt that is larger than (2^63 -1)', function () {
+    const maxIntPlusOne = { test: 2n ** 63n };
+    const serializedMaxIntPlusOne = BSON.serialize(maxIntPlusOne);
+    const expectedResultForMaxIntPlusOne = byteUtils.bufferFromHexArray([
+      '12', // int64
+      Buffer.from('test\x00', 'utf8').toString('hex'),
+      '0000000000000080'
+    ]);
+    expect(serializedMaxIntPlusOne).to.deep.equal(expectedResultForMaxIntPlusOne);
+  });
+
+  it('Correctly serializes BigInts at the edges of the valid range [-2^63, 2^63 - 1]', function () {
+    const maxPositiveInt64 = { test: 2n ** 63n - 1n };
+    const serializedMaxPositiveInt64 = BSON.serialize(maxPositiveInt64);
+    const expectedSerializationForMaxPositiveInt64 = byteUtils.bufferFromHexArray([
+      '12', // int64
+      Buffer.from('test\x00', 'utf8').toString('hex'),
+      'ffffffffffffff7f'
+    ]);
+    expect(serializedMaxPositiveInt64).to.deep.equal(expectedSerializationForMaxPositiveInt64);
+
+    const minPositiveInt64 = { test: -(2n ** 63n) };
+    const serializedMinPositiveInt64 = BSON.serialize(minPositiveInt64);
+    const expectedSerializationForMinPositiveInt64 = byteUtils.bufferFromHexArray([
+      '12', // int64
+      Buffer.from('test\x00', 'utf8').toString('hex'),
+      '0000000000000080'
+    ]);
+    expect(serializedMinPositiveInt64).to.deep.equal(expectedSerializationForMinPositiveInt64);
+  });
+
+  it('Correctly truncates a BigInt that is larger than a 64-bit int', function () {
+    const testDoc = { test: 2n ** 64n + 1n };
+    const serializedDoc = BSON.serialize(testDoc);
+    const expectedSerialization = byteUtils.bufferFromHexArray([
+      '12', //int64
+      Buffer.from('test\x00', 'utf8').toString('hex'),
+      '0100000000000000'
+    ]);
+    expect(serializedDoc).to.deep.equal(expectedSerialization);
+  });
+
+  it('Calls toBSON on a BigInt value if it exists', function () {
+    BigInt.prototype.toBSON = () => `hello`;
```

Comment:
If a error occurs within this test we'll have left a toBSON method on the BigInt prototype which could effect other tests in positive or negative ways. So to avoid that, adding/removing this method should be done in a beforeEach/afterEach hook. Not all bigint tests need to be in this file it may make more sense to test this with the other concerns of toBSON. But positioning of tests isn't critical as long as they're written in a safe way (don't impact other tests) and communicate intention.

Reply:
I see what you mean. Also, it seems that there is already a test for this in `test/node/to_bson_test.js:133-186`, so I can just go ahead and delete this test here altogether.